src/mjolnir/adminbuilder.cc

```diff
@@ -12,206 +12,175 @@
 
 #include "config.h"
 
-/* Need to know which geos version we have to work out which headers to include */
-#include <geos/version.h>
-
-#define USE_UNSTABLE_GEOS_CPP_API
-#if GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR >= 8
-#include <geos/geom/CoordinateArraySequence.h>
-#else
-#include <geos/geom/CoordinateSequenceFactory.h>
-#endif
-#include <geos/geom/Geometry.h>
-#include <geos/geom/GeometryFactory.h>
-#include <geos/geom/LineString.h>
-#include <geos/geom/LinearRing.h>
-#include <geos/geom/MultiLineString.h>
-#include <geos/geom/MultiPolygon.h>
-#include <geos/geom/Point.h>
-#include <geos/geom/Polygon.h>
-#include <geos/io/WKTReader.h>
-#include <geos/io/WKTWriter.h>
-#if GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR >= 9
-#include <geos/operation/linemerge/LineMerger.h>
-#else
-#include <geos/opLinemerge.h>
-#endif
-#include <geos/util/GEOSException.h>
-
-#include <boost/optional.hpp>
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
 #include <boost/property_tree/ptree.hpp>
 
-using namespace geos::geom;
-using namespace geos::io;
-using namespace geos::util;
-using namespace geos::operation::linemerge;
+using polar_coordinate_system_t = boost::geometry::cs::geographic<boost::geometry::degree>;
+using point_t = boost::geometry::model::d2::point_xy<double, polar_coordinate_system_t>;
+using polygon_t = boost::geometry::model::polygon<point_t>;
+using multipolygon_t = boost::geometry::model::multi_polygon<polygon_t>;
+
+// in order to work with boost points inside stl containers
+namespace std {
+template <> struct hash<point_t> {
+  inline size_t operator()(const point_t& p) const {
+    return (uint64_t(p.x() * 1e7 + 180 * 1e7) << 32) | (uint64_t(p.y() * 1e7 + 90 * 1e7));
+  }
+};
+template <> struct equal_to<point_t> {
+  inline bool operator()(const point_t& a, const point_t& b) const {
+    return a.x() == b.x() && a.y() == b.y();
+  }
+};
+} // namespace std
 
 // For OSM pbf reader
 using namespace valhalla::mjolnir;
 using namespace valhalla::baldr;
-
 using namespace valhalla::midgard;
 
 namespace {
 
-struct polygondata {
-  Polygon* polygon;
-  LinearRing* ring;
-  double area;
-  int iscontained;
-  unsigned containedbyid;
-};
-
-int polygondata_comparearea(const void* vp1, const void* vp2) {
-  const polygondata* p1 = (const polygondata*)vp1;
-  const polygondata* p2 = (const polygondata*)vp2;
+/**
+ * Converts a series of a linestrings into one or more polygons (rings) by connecting contiguous
+ * ones until a ring is formed
+ * @param lines the unmerged lines
+ * @param line_lookup a multi map that lets one easily
+ * @return
+ */
+std::vector<std::vector<point_t>> line_merge(std::vector<std::vector<point_t>> lines,
```

Comment:
this line merging to make rings algorithm turned out a hell of a lot simpler than the one i implemented in the isochrones code.. im not sure why i didnt do it this way over there as well.. itd be interesting to see if younger kevin was right to have that be super complicated or if this simpler approach would be better. im sure it comes down to the nuances of map vs multimap (both unordered) and list vs vector (ie non contiguous mem vs reallocs)